File: backend/src/main/java/io/github/xpakx/ladder/entity/dto/UserWithData.java
package io.github.xpakx.ladder.entity.dto;

import lombok.Getter;
import lombok.Setter;

import java.util.List;

@Setter
@Getter
public class UserWithData {
    private Integer id;
    private String username;
    private boolean projectCollapsed;
    private List<ProjectDetails> projects;
    private List<TaskDetails> tasks;
    private List<LabelDetails> labels;
    private List<HabitDetails> habits;
    private List<HabitCompletionDetails> todayHabitCompletions;
}

File: frontend/src/app/entity/user-with-data.ts
import { CollabProjectData } from "../project/dto/collab-project-data";
import { CollabTaskDetails } from "../task/dto/collab-task-details";
import { FilterDetails } from "../filter/dto/filter-details";
import { HabitCompletionDetails } from "../habit/dto/habit-completion-details";
import { HabitDetails } from "src/app/habit/dto/habit-details";
import { LabelDetails } from "../label/dto/label-details";
import { ProjectDetails } from "src/app/project/dto/project-details";
import { TaskDetails } from "../task/dto/task-details";

export interface UserWithData {
    username: string;
    id: number;
    projectCollapsed: boolean;
    projects: ProjectDetails[];
    tasks: TaskDetails[];
    labels: LabelDetails[];
    habits: HabitDetails[];
    todayHabitCompletions: HabitCompletionDetails[];
    filters: FilterDetails[];
    collabs: CollabProjectData[];
    collabTasks: CollabTaskDetails[];
}